pkg/trace/sampler/remote_rates.go

```diff
@@ -64,7 +70,10 @@ func newRemoteRates(maxTPS float64) *RemoteRates {
 
 func (r *RemoteRates) loadNewConfig(new *pbgo.ConfigResponse) error {
 	log.Debugf("fetched config version %d from remote config management", new.ConfigDelegatedTargetVersion)
-	tpsTargets := make(map[Signature]float64, len(r.tpsTargets))
+	r.mu.RLock()
+	size := len(r.tpsTargets)
+	r.mu.RUnlock()
+	tpsTargets := make(map[Signature]pb.TargetTPS, size)
```

Comment:
It works, because `trace-dynamic-sampling-rates` pushes the following struct, right?
```
type TargetTPS struct {
	Service   string  `msgpack:"0"`
	Env       string  `msgpack:"1"`
	Value     float64 `msgpack:"2"`
	Rank      uint32  `msgpack:"3"`
	Mechanism uint32  `msgpack:"4"`

	ServiceOld   string  `msgpack:"service"`
	EnvOld       string  `msgpack:"env"`
	ValueOld     float64 `msgpack:"value"`
}
```

Reply:
yes, remote will send a backward compatible payload until the trace-agent is deployed everywhere